api/state.go

```diff
@@ -42,6 +44,7 @@ func (st *state) Login(tag names.Tag, password, nonce string, macaroons []macaro
 		Credentials: password,
 		Nonce:       nonce,
 		Macaroons:   macaroons,
+		CLIArgs:     utils.CommandString(os.Args...),
```

Comment:
We should check the back end version I suspect - we may need to go to Login v4?
We typically avoid sending data to the apiserver which it is not expecting.

Reply:
Leaving the API at v3 was in the original spec - the thinking is:
> Since the CLI args are entirely optional, and this isn't changing the behaviour of the Login method, I propose that we keep the Facade version at 3, and just add the CLIArgs to the params.LoginRequest structure. If a more modern client calls an older API server the extra argument is thrown away silently. An older client will not pass the value through, so the value will be empty. Both of these situations are fine.

Reply:
And therein lies the issue - the modern client *thinks* the CLI args are being recorded and they are not, since the extra arg is silently discarded.

Reply:
But is that a problem? It's not the client's concern whether the controller is doing audit logging, it's the controller's. If someone is running a controller that doesn't do audit logging, the more modern client shouldn't refuse to work.

Reply:
For future reference, after more discussion I did try revving the Admin API. Unfortunately bumping the version specified for the Admin.Login call to 4 in the client meant that it couldn't connect to older controllers that only support v3 for Admin. The client can't use the usual version negotiation here because the LoginResult is what indicates supported versions for facades, so the solution would be to try v4 then fallback to v3. That's overkill in this case, so I'm leaving this as is.

Reply:
I think that it is OK. From the server side it is optional, and if a client sends it and the server isn't expecting it, nothing untoward happens.